chore(worktree): clone venv and node_modules with copy-on-write

webjunkie opened the pull request:
**What:** Use APFS clones (macOS) or reflinks (Linux) to share `.flox/cache/venv` and `node_modules` between worktrees.

**Why:** Saves ~4GB disk space per worktree while still providing independent copies that can diverge. CoW clones share data blocks until modified.

**Details:**
- Clones `.flox/cache/venv` (~1.8GB) — the Python venv managed by flox/uv
- Clones `node_modules` (~2.5GB) — pnpm dependencies
- If CoW not available (cross-volume, unsupported fs), skips with a message
- Small files (.envrc, .env, .vscode/settings.json) fall back to regular copy

webjunkie committed:
chore(worktree): clone venv and node_modules with copy-on-write

Use APFS clones (macOS) or reflinks (Linux) to share .flox/cache/venv
and node_modules between worktrees. Saves disk space while providing
each worktree with its own copy that can diverge independently.

If CoW is not available (cross-volume, unsupported filesystem),
the large directories are skipped — flox activate and pnpm install
will create them.

haacked commented on bin/posthog-worktree:
```diff
@@ -39,6 +39,31 @@ done
+    else
+        if cp --reflink=auto "${cp_args[@]}" "$source_path" "$destination_path" 2>/dev/null; then
+            return 0
+        fi
```
`--reflink=auto` silently falls back to regular copy on non-CoW filesystems (ext4), so this always succeeds and returns 0. The `require_cow=true` logic below never triggers on Linux.
 
To actually require CoW, use `--reflink=always` when `require_cow` is `true`:

```bash
if [[ "$require_cow" == "true" ]]; then
    cp --reflink=always "${cp_args[@]}" "$source_path" "$destination_path" 2>/dev/null
else
    cp --reflink=auto "${cp_args[@]}" "$source_path" "$destination_path" 2>/dev/null
fi
```

Otherwise Linux users on ext4 get silent 4GB copies instead of the intended skip.

webjunkie committed:
Merge remote-tracking branch 'origin/master' into chore/worktree-cow-clone

webjunkie committed:
fix(worktree): use --reflink=always on Linux when CoW required

--reflink=auto silently falls back to regular copy on ext4, so the
require_cow logic never triggered. Use --reflink=always to properly
fail when CoW is not available.

webjunkie committed:
Merge branch 'master' into chore/worktree-cow-clone

haacked reviewed (approved):
Nice work! I'm hoping this saves me a ton of disk space!
